packages/runtime/src/templates/server.ts

```diff
@@ -118,6 +119,13 @@ const getNetlifyNextServer = (NextServer: NextServerType) => {
       }
     }
 
+    // eslint-disable-next-line class-methods-use-this, require-await
+    async runMiddleware(): Promise<{ finished: boolean }> {
```

Comment:
Just linking to Next.js server's implementation for reference. https://github.com/vercel/next.js/blob/1286e145b058538e47b11cc1ec813a686cee1f68/packages/next/src/server/next-server.ts#L1474-L1589